java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -1296,6 +1346,86 @@ public ColumnVector castTo(DType type) {
     return new ColumnVector(castTo(getNativeView(), type.typeId.getNativeId(), type.getScale()));
   }
 
+  /**
+   * This method takes in a nested type and replaces its children with the given views
+   * Note: Make sure the numbers of rows in the leaf node are the same as the child replacing it
+   * otherwise the list can point to elements outside of the column values.
+   *
+   * Note: this method returns a ColumnView that won't live past the ColumnVector that it's
+   * pointing to.
+   *
+   * Ex: List<Int> list = col{{1,3}, {9,3,5}}
+   *
+   * validNewChild = col{8, 3, 9, 2, 0}
+   *
+   * list.replaceChildrenWithViews(1, validNewChild) => col{{8, 3}, {9, 2, 0}}
+   *
+   * invalidNewChild = col{3, 2}
+   * list.replaceChildrenWithViews(1, invalidNewChild) => col{{3, 2}, {invalid, invalid, invalid}}
+   *
+   * invalidNewChild = col{8, 3, 9, 2, 0, 0, 7}
+   * list.replaceChildrenWithViews(1, invalidNewChild) => col{{8, 3}, {9, 2, 0}} // undefined result
+   */
+  public ColumnView replaceChildrenWithViews(int[] indices,
```

Comment:
I thought the point of the new `ColumnView` constructors is to not have this specific API in cudf?

Reply:
I wanted to keep this as a convenience API to replace children. 

I can remove if you want

Reply:
It's fine for now, we can use this to help hide the string/list differences for the short-term.